docs/src/pages/concepts/building-forms.mdx

```diff
@@ -0,0 +1,98 @@
+Accessibility best practices should be followed, such as providing clear, descriptive labels and ensuring that form fields are appropriately labeled and structured for screen readers.
+
+Here you can see how to layout a classic form with Marigold's `<TextField>` and `<Select>` could be defined.
+
+If you want to see more examples of forms with Marigold, check out our recipes: [ Marigold Form Recipes](/recipes/form-recipes)
+
+```tsx preview file=concepts/layout-forms.demo.tsx
+
+```
+
+## Controlled or uncontrolled Components?
+
+In React, there are two main types of form components: controlled and uncontrolled.
+Controlled components are components which are controlled by the React state. Any changes to the value will be updated via the state variable. This happens when the user interacts with a component and the `onChange` event handler is called.
+Uncontrolled components don't need the state to update, because you can give them a `defaultValue` or `defaultChecked` property.
+
+When the state or props change, React will automatically re-render the component with its new data. This is called [data-binding](https://www.joshwcomeau.com/react/data-binding/). Components recive data from its parents component and update their output in response to changes in that data by using state and props, this allows to have a dynamic and reactive user interface.
+
+## Interactive Forms
+
+Interactve forms can have several features like error validation, real-time feedback (password-stregth indicator), logic (hide fields if a several field is checked etc.) or autocomplete.
+All these features help to make interactive fields more user friendly and more efficient.
+
+### Error Handling
+
+Marigold's form components comes with error props. You can use them to show instead of a helptext an error message. For that the only thing to do is add the prop and make a check.
+
+In this example below you can have an example on how to check if the E-Mail adress is correctly filled.
+Try out and type something!
+
+```tsx preview file=concepts/error-handling.demo.tsx
+
+```
+
+### Logical Interaction
+
+You can have interactive forms which contains logical aspects. In this example you can interact with the `<Select>` and a new component `<Checkbox>` will appear. You can try out and switch the value of the select field.
+
+```tsx preview file=concepts/logical-interaction.demo.tsx
+
+```
+
+## Submitting a form
+
+In this example you can see how a form with all its fields can be submitted. The submitted data will be displayed in an alert message.
+
+For validate the errors we have used [`Zod`](https://zod.dev/). It is a simple tool for building schemas, which can be validated.
+
+```tsx preview file=concepts/building-forms-zod.demo.tsx
+
+```
+
+## Full example of submit a form with React Hook Form
```

Comment:
```suggestion
## Submit a form with React Hook Form
```

Reply:
Maybe we need to give more context here? Something like "when forms get more complex"?

Reply:
"Handling complex forms and form state"?